MetaMask/metamask-extension
# Fix for connected sites throwing error if name is null.

## Explanation


There's an issue with connected sites where if the name attr of site metadata is null, extension throws an exception,
below is an example of a connected site with name=null from `state.metamask.subjectMetadata`

```
chrome-extension://ahbgkigagdiopfeihifjapallnjepehi]()": {
        "origin": "[chrome-extension://ahbgkigagdiopfeihifjapallnjepehi]()",
        "extensionId": "ahbgkigagdiopfeihifjapallnjepehi",
        "subjectType": "extension",
        "iconUrl": null,
        "name": null
      }
```

The fix is to check name before calling name.length. 

## More information



## Screenshots/Screencaps


### Before

<img width="829" alt="Screenshot 2022-03-15 at 22 21 18" src="https://user-images.githubusercontent.com/44811/158445803-7d6b81bb-7d9b-4b1d-bc6d-7fba3a7f9d9c.png">


### After

<img width="378" alt="Screenshot 2022-03-15 at 22 21 58" src="https://user-images.githubusercontent.com/44811/158445862-3241b6fd-ad0b-4a1e-ba04-416853b4e295.png">

## Manual testing steps

I can't reproduce this because I am not sure how to add extensions with null names, but I can simulate it.

Goto [here](https://github.com/MetaMask/metamask-extension/blob/99da6a9283d257c10151c7de267c9c219ef48d0d/ui/selectors/permissions.js#L92)

change `name` to `name: null`

Run Metamask to see the error.
